### PR TITLE
Typeahead MinLength=0

### PR DESCRIPTION
**Pull request by rbaarsma**
Made it possible to use the TypeaheadMinLength feature with value=0. This will show the entire list of options as soon as the 'focus' event is fired and also when you remove all text.

Would add feature for https://github.com/valor-software/ng2-bootstrap/issues/413

### CODE REVIEW

**Comment by rbaarsma** [comment]
also fixes duplicate issue #187

File: components/typeahead/typeahead.directive.ts [hunk 1]
```diff
@@ -266,6 +282,13 @@ export class Typeahead implements OnInit {
       return;
     }
 
+    if (!this.cd.model) {
+      for (let i = 0; i < this.typeahead.length; i++) {
```
Comment (valorkin) [inline components/typeahead/typeahead.directive.ts]: i < Math.min(this.typeaheadOptionsLimit, this.typeahead.length)

File: components/typeahead/typeahead.directive.ts [hunk 2]
```diff
@@ -100,6 +100,22 @@ export class Typeahead implements OnInit {
...
+  @HostListener('focus', ['$event.target'])
+  protected onFocus():void {
+    if (this.typeaheadMinLength === 0) {
+      this.typeaheadLoading.emit(true);
+
+      if (this.typeaheadAsync === true) {
+        this.debouncer();
+      }
+
+      if (this.typeaheadAsync === false) {
```
Comment (valorkin) [inline components/typeahead/typeahead.directive.ts]: `if (!this.typeaheadAsync)` could be undefined

**Comment by valorkin** [comment]
@rbaarsma looks good to me!
please rebase on latest and apply\validate my comments :)

**Comment by valorkin** [comment]
@rbaarsma I have reread your issue about showing all options,
than special `-1` value should be added to `typeaheadOptionsLimit`
treated in code and mentioned in doc

**Comment by rbaarsma** [comment]
@valorkin I have implemented your comments. I'm not sure what you mean with the special `-1` value, but what I understand from https://github.com/valor-software/ng2-bootstrap/issues/187 is that typeaheadMinLength 0 was a feature of the angular1 bootstrap as well. Makes sense to keep it the same and min-length of 0 makes sense. I'd leave it as it is.

As for rebasing it on latest. Unfortunately I'm pretty new to working with pull requests with open source projects and I don't want to do anything wrong. I did update my branch to the latest version. I thought you could simply merge it like this, but if I need to do something else can you please tell me what to do exactly to rebase :)?

**Comment by valorkin** [comment]
ok, looks good to me!
please describe this option in doc
PS: I will squash on merge
PSS: God bless @github for this option :D

**Comment by rbaarsma** [comment]
@valorkin updated readme. I think you can merge it now :smile:

**Comment by valorkin** [comment]
Great! Merged

**Pull request merged**